src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;script&gt; elements that supports fallback src paths.
+    /// </summary>
+    public class ScriptTagHelper : TagHelper
+    {
+        private const string FallbackSrcAttributeName = "asp-fallback-src";
+        private const string FallbackTestMethodAttributeName = "asp-fallback-test";
+        private const string SrcAttributeName = "src";
+
+        // NOTE: All attributes are required for the LinkTagHelper to process.
+        private static readonly string[] RequiredAttributes = new[]
+        {
+            FallbackSrcAttributeName,
+            FallbackTestMethodAttributeName,
+        };
+
+        /// <summary>
+        /// The URL of a Script tag to fallback to in the case the primary one fails (as specified in the src
+        /// attribute).
+        /// </summary>
+        [HtmlAttributeName(FallbackSrcAttributeName)]
+        public string FallbackSrc { get; set; }
+
+        /// <summary>
+        /// The script method defined in the primary script to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestMethodAttributeName)]
+        public string FallbackTestMethod { get; set; }
+
+        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
+        [Activate]
+        protected internal ILogger<ScriptTagHelper> Logger { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            // TODO: Consider using the RazorStringWriter instead of a StringBuilder, because Script tags may contain large bodies.
+
+            if (!context.AllRequiredAttributesArePresent(RequiredAttributes, Logger))
+            {
+                if (Logger.IsEnabled(LogLevel.Verbose))
+                {
+                    Logger.WriteVerbose("Skipping processing for {0} {1}", nameof(ScriptTagHelper), context.UniqueId);
+                }
+
+                return;
+            }
+
+            var content = new StringBuilder();
+
+            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+
+            // We've taken over rendering here so prevent the element rendering the outer tag
+            output.TagName = null;
+
+            // Rebuild the <script /> tag.
+            content.Append("<script");
+            foreach (var attribute in output.Attributes)
+            {
+                content.AppendFormat(CultureInfo.InvariantCulture, " {0}='{1}'", attribute.Key, attribute.Value);
```

Comment:
this is completely intentional. The code and the html look much simpler and do not require escaping. Unfortunately we don't know what the user used, so we can't accommodate his style, we could do that when that knowledge is captured.

@dougbu the single quote statement is a stronger statement than the one @NTaylorMullen made, that might sway me in that direction. Ultimately we want to just follow what the user wrote in the original tag in the .cshtml tag though.

@dougbu not sure I buy this 100% and was following `LinkTagHelper`. I'll take a second look into `TagBuilder`, also note this code builds a brand new tag after the existing tag.